networkx/algorithms/cycles.py

```diff
@@ -585,11 +585,13 @@ def chordless_cycles(G, length_bound=None):
     # Nodes with loops cannot belong to longer cycles.  Let's delete them here.
     # also, we implicitly reduce the multiplicity of edges down to 1 in the case
     # of multiedges.
+    self_loop_nodes = set(nx.nodes_with_selfloops(G))
+    Gless = G.subgraph(set(G) - self_loop_nodes).copy()
     if directed:
-        F = nx.DiGraph((u, v) for u, Gu in G.adj.items() if u not in Gu for v in Gu)
+        F = nx.DiGraph((u, v) for u, Gu in Gless.adj.items() if u not in Gu for v in Gu)
```

Comment:
This change to `Gless` is not needed. The phrase `if u not in Gu` already removes all nodes with self-loops.

I would suggest not creating `Gless`. Instead use this line (and the undirected case line just below) to construct `F` without the nodes that have self-loops. Then in the rest of the code use `F` instead of `Gless`.  Does this work?  It seems to avoid the extra work in lines 588 and 589.